modules/swagger-codegen-common-feign/src/main/java/io/swagger/codegen/common/feign/SwaggerFeignApiClient.java

```diff
@@ -0,0 +1,34 @@
+package io.swagger.codegen.common.feign;
+
+import feign.Feign;
+import feign.RequestInterceptor;
+
+import java.util.Map;
+
+
+public interface SwaggerFeignApiClient {
```

Comment:
This is the common interface that all `ApiClient` classes generated for feign-based Java clients will implement. It's a reasonable start with minimal dependencies (only feign) that defines everything we need for our use case.